website/docs/language/mitigate-risk/validate.mdx

```diff
@@ -0,0 +1,248 @@
+- Validate the behavior of your infrastructure as a whole
+- Verify infrastructure configuration without blocking operations
+- Perform [continuous validation in HCP Terraform](#continuous-validation-in-hcp-terraform)
+
+The following example uses a `check` block to verify the Terraform website is healthy.
+
+```hcl
+check "health_check" {
+  data "http" "terraform_io" {
+    url = "https://www.terraform.io"
+  }
+
+  assert {
+    condition = data.http.terraform_io.status_code == 200
+    error_message = "${data.http.terraform_io.url} returned an unhealthy status code"
+  }
+}
+```
+
+If the website's endpoint returns a `200` status code, then the website is healthy and the check passes. Unlike other ways of validating your configuration, `check` blocks do not block operations. If the assertion evaluates to `false`, Terraform throws a warning that includes the result of the `error_message` expression and continues the operation.
+
+For more details, refer to the [`check` configuration reference](/terraform/language/block/check).
+
+## Continuous validation in HCP Terraform
+
+If you enable health checks on a workspace, HCP Terraform continuously validates any `check` blocks, preconditions, and postconditions in your workspace's configuration to regularly verify your infrastructure. For example, you can use a `check` block to continuously monitor the validity of an API gateway certificate. Continuous validation alerts you when the condition fails, so you can update the certificate and avoid errors the next time you want to update your infrastructure.
+
+Refer to [Continuous validation](/terraform/cloud-docs/workspaces/health#continuous-validation) for more details.
+
+## Order of validation
+
+Terraform validates different aspects of your configuration as early as it can. Generally, Terraform executes evaluations in the following order:
+
+1. Terraform executes input variable validations immediately, before generating a plan.
+1. Terraform executes preconditions after generating a plan, but before creating the resource, data source, or output.
+1. Terraform executes postconditions after planning and applying changes.
+1. Terraform executes checks at the end of plan and apply operations, and whenever health assessments run on a workspace in HCP Terraform.
+
+
+![Order of operations for when validations run](/img/docs/validation-order-of-operations-dark.jpg#light-theme-only)
+
+![Order of operations for when validations run](/img/docs/validation-order-of-operations-dark.jpg#dark-theme-only)
+
+The precise order that Terraform executes `check` blocks, preconditions, and postconditions can depend on whether Terraform knows a condition's value before or after applying your configuration. If the relevant value is available before an apply operation, then Terraform performs the validation during the planning phase. For example, if Terraform knows a resource's image ID during planning, it can execute any validations that rely on that ID.
```

Comment:
```suggestion
The precise order that Terraform executes `check` blocks, preconditions, and postconditions can depend on whether Terraform has information about a condition's value before or after applying your configuration. If the relevant value is available before an apply operation, then Terraform performs the validation during the planning phase. For example, if Terraform logs a resource's image ID during planning, it can execute any validations that rely on that ID.
```
We need to figure out more precise language about what it means for TF to "know" something so that we can edit out personification from the docs. Is the information logged internally? Is it captured in state? Is it cached or written to a temporary file?

Reply:
Good point! I think it's different at different points, "logs" isn't the _quite_ right word because what matters is if TF has access to the value. Accepting with a slight tweak:

```suggestion
The precise order that Terraform executes `check` blocks, preconditions, and postconditions can depend on whether Terraform has information about a condition's value before or after applying your configuration. If the relevant value is available before an apply operation, then Terraform performs the validation during the planning phase. For example, if Terraform has access to a resource's image ID during planning, it can execute any validations that rely on that ID.
```